enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/MobileTokenController.java

```diff
@@ -118,6 +120,78 @@ public ObjectResponse<OperationListResponse> operationList(@Parameter(hidden = t
+     */
+    @PostMapping("/operation/detail")
+    @PowerAuthToken(signatureType = {
+            PowerAuthSignatureTypes.POSSESSION,
+            PowerAuthSignatureTypes.POSSESSION_BIOMETRY,
+            PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE,
+            PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE_BIOMETRY
+    })
+    public ObjectResponse<Operation> getOperationDetail(@RequestBody ObjectRequest<OperationDetailRequest> request,
+                                                        @Parameter(hidden = true) PowerAuthApiAuthentication auth,
+                                                        @Parameter(hidden = true) Locale locale) throws MobileTokenException, MobileTokenConfigurationException {
+        try {
+            if (auth != null) {
+                final String operationId = request.getRequestObject().getId();
+                final String language = locale.getLanguage();
+                final String userId = auth.getUserId();
+                final Operation response = mobileTokenService.getOperationDetail(operationId, language, userId);
+                final Date currentTimestamp = new Date();
+                return new MobileTokenResponse<>(response, currentTimestamp);
+            } else {
+                throw new MobileTokenAuthException();
+            }
+        } catch (PowerAuthClientException e) {
+            logger.error("Unable to call upstream service.", e);
+            throw new MobileTokenAuthException();
+        }
+    }
+
+    /**
+     * Claim operation for a user.
+     *
+     * @param auth Authentication object.
+     * @param locale Locale.
+     * @return List of pending operations.
+     * @throws MobileTokenException In the case error mobile token service occurs.
+     * @throws MobileTokenConfigurationException In the case of system misconfiguration.
+     */
+    @PostMapping("/operation/detail/claim")
+    @PowerAuthToken(signatureType = {
+            PowerAuthSignatureTypes.POSSESSION,
+            PowerAuthSignatureTypes.POSSESSION_BIOMETRY,
+            PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE,
+            PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE_BIOMETRY
+    })
+    public ObjectResponse<Operation> claimOperation(@RequestBody ObjectRequest<OperationDetailRequest> request,
+                                                    @Parameter(hidden = true) PowerAuthApiAuthentication auth,
+                                                    @Parameter(hidden = true) Locale locale) throws MobileTokenException, MobileTokenConfigurationException {
+        try {
+            if (auth != null) {
+                final String operationId = request.getRequestObject().getId();
+                final String language = locale.getLanguage();
+                final String userId = auth.getUserId();
+                final Operation response = mobileTokenService.claimOperation(operationId, language, userId);
```

Comment:
🤔 I was about to introduce a template method using lambdas, but since service methods throw check exceptions, let it be.